frontend/src/components/Dialogs/InputDialogAttachUrl.vue

```diff
@@ -1,73 +1,121 @@
+  <div v-if="'name' in userInput && highlight && !name" class="required-message" data-test="required">
+    *required
+  </div>
+  <input placeholder="Paste a link (required)" v-model="url" data-test="link-input"
+    :class="{ 'highlight': highlight && !url }" />
+  <div v-if="highlight && !url" class="required-message" data-test="required">
+    *required
+  </div>
+  <textarea v-if="'comments' in userInput" class="comments" placeholder="Comments" v-model="comments"
+    data-test="comments-text-area">
   </textarea>
 </template>
 
+
 <script>
 export default {
   name: 'input-dialog-attach-url',
-  emits: ['update:userInput'],
+  emits: ['update:userInput', 'update:allRequiredFieldsFilled', 'validationDone'],
   props: {
     userInput: {
       type: Object,
       required: true,
     },
+    runValidation: {
+      type: Boolean,
+    },
+  },
+  data() {
+    return {
+      highlight: false,
+    };
   },
   computed: {
+    requiredFields() {
+      return this.userInput.requiredFields || [];
+    },
     name: {
       get() {
         return this.userInput.name;
       },
       set(value) {
-        const input = this.userInput;
-        input['name'] = value;
-        this.$emit('update:userInput', input);
+        this.updateUserInput('name', value);
       },
     },
     url: {
       get() {
         return this.userInput.data;
       },
       set(value) {
-        const input = this.userInput;
-        input['data'] = value;
-        this.$emit('update:userInput', input);
+        this.updateUserInput('data', value);
       },
     },
     comments: {
       get() {
         return this.userInput['comments'];
       },
       set(value) {
-        const input = this.userInput;
-        input['comments'] = value;
-        this.$emit('update:userInput', input);
+        this.updateUserInput('comments', value);
       },
     },
+    allRequiredFieldsFilled() {
+      return this.requiredFields.every((field) => !!this[field] && this[field].length > 0);
+    },
+  },
+  watch: {
+    runValidation(currentValidationState, previousValidationState) {
+      if (currentValidationState !== previousValidationState && currentValidationState === true) {
+        this.validateRequiredFields();
+        this.$emit('validationDone');
+      }
+    },
+  },
+  methods: {
+    confirmClicked() {
+      this.validateRequiredFields();
+    },
+    updateUserInput(field, value) {
+      const input = this.userInput;
+      input[field] = value;
+      this.$emit('update:userInput', input);
+    },
+    validateRequiredFields() {
+      const allFieldsFilled = this.requiredFields.every((field) => !!this[field] && this[field].length > 0);
+      if (!allFieldsFilled) {
+        this.highlightRequiredFields();
+      } else {
+        this.$emit('update:allRequiredFieldsFilled', allFieldsFilled);
+      }
+    },
+    highlightRequiredFields() {
+      this.highlight = true;
+    },
   },
 };
 </script>
 
+
 <style>
+.highlight {
```

Comment:
Since the styling is going to be utilized in more areas than one, lets move it to the `rosalution.css ` and rename it to be more descriptive, like 'invalid-input-highlight'.

Reply:
Made the requested change, and awaiting the additional change requests